src/Types/VersionedModuleSource.cs

```diff
@@ -100,9 +164,46 @@ versionedModuleSource.ModuleCase switch
 /// <param name="Source">Source code of module</param>
 public sealed record ModuleV0(byte[] Source) : VersionedModuleSource(Source)
 {
+    internal override uint GetVersion() => 0;
+
     internal static ModuleV0 From(Grpc.V2.VersionedModuleSource.Types.ModuleSourceV0 moduleSourceV0) =>
         new(moduleSourceV0.Value.ToByteArray());
 
+    /// <summary>
+    /// Creates a WASM-module from byte array.
+    /// </summary>
+    /// <param name="source">WASM-module as a byte array.</param>
+    /// <exception cref="ArgumentException">The length of the supplied module exceeds "MaxLength".</exception>
+    public static ModuleV0 From(byte[] source)
+    {
+        if (source.Length > MaxLength)
+        {
+            throw new ArgumentException(
+                $"Size of a data is not allowed to exceed {MaxLength} bytes."
+            );
+        }
+
+        return new ModuleV0(source.ToArray());
```

Comment:
Do we need to copy the array?

Reply:
I would think so? Otherwise, mutating the provided array would mutate the module. That doesn't seem intuitive to me.

Reply:
I don't expect users will mutate these, and avoiding the copy would be good.
Maybe just document that this assumes the bytes are never mutated, otherwise the user will need to copy before calling.

Also, since it is an array, they can only change the value of the bytes and not add or remove anything, meaning the above check of the length is still kept valid in the case they mutate.